app/src/util/api.ts

```diff
@@ -2,10 +2,18 @@ import createHttpError from "http-errors";
 import { NextResponse } from "next/server";
 import { ZodError } from "zod";
 
+import { db } from "@/models";
+
 export function apiHandler(handler: (req: Request) => Promise<NextResponse>) {
   return async (req: Request) => {
     try {
+      if (!db.initialized) {
+        await db.initialize();
+      }
+
       // TODO JWT authentication logic here
+      // await jwtMiddleware(req);
```

Comment:
So, we'll need both session and JWT auth.

Reply:
Yes, to support a proper (immediate) logout I would suggest using redis for storing the sessions.